Model/Api/CreateOrder.php

```diff
@@ -0,0 +1,694 @@
+    const E_BOLT_ORDER_ALREADY_EXISTS = 2001002;
+    const E_BOLT_CART_HAS_EXPIRED = 2001003;
+    const E_BOLT_ITEM_PRICE_HAS_BEEN_UPDATED = 2001004;
+    const E_BOLT_ITEM_OUT_OF_INVENTORY = 2001005;
+    const E_BOLT_DISCOUNT_CANNOT_APPLY = 2001006;
+    const E_BOLT_DISCOUNT_CODE_DOES_NOT_EXIST = 2001007;
+    const E_BOLT_SHIPPING_EXPIRED = 2001008;
+
+    /**
+     * @var HookHelper
+     */
+    private $hookHelper;
+
+    /**
+     * @var OrderHelper
+     */
+    private $orderHelper;
+
+    /**
+     * @var LogHelper
+     */
+    private $logHelper;
+
+    /**
+     * @var Request
+     */
+    private $request;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @var Response
+     */
+    private $response;
+
+    /**
+     * @var ConfigHelper
+     */
+    private $configHelper;
+
+    /**
+     * @var CartHelper
+     */
+    private $cartHelper;
+
+    /**
+     * @var UrlInterface
+     */
+    private $url;
+
+    /**
+     * @var BackendUrl
+     */
+    private $backendUrl;
+
+    /**
+     * @var StockRegistryInterface
+     */
+    private $stockRegistry;
+
+    /**
+     * @var SessionHelper
+     */
+    private $sessionHelper;
+
+    /**
+     * @param HookHelper             $hookHelper
+     * @param OrderHelper            $orderHelper
+     * @param CartHelper             $cartHelper
+     * @param LogHelper              $logHelper
+     * @param Request                $request
+     * @param Bugsnag                $bugsnag
+     * @param Response               $response
+     * @param UrlInterface           $url
+     * @param BackendUrl             $backendUrl
+     * @param ConfigHelper           $configHelper
+     * @param StockRegistryInterface $stockRegistry
+     * @param SessionHelper          $sessionHelper
+     */
+    public function __construct(
+        HookHelper $hookHelper,
+        OrderHelper $orderHelper,
+        CartHelper $cartHelper,
+        LogHelper $logHelper,
+        Request $request,
+        Bugsnag $bugsnag,
+        Response $response,
+        UrlInterface $url,
+        BackendUrl $backendUrl,
+        ConfigHelper $configHelper,
+        StockRegistryInterface $stockRegistry,
+        SessionHelper $sessionHelper
+    ) {
+        $this->hookHelper = $hookHelper;
+        $this->orderHelper = $orderHelper;
+        $this->logHelper = $logHelper;
+        $this->request = $request;
+        $this->bugsnag = $bugsnag;
+        $this->response = $response;
+        $this->configHelper = $configHelper;
+        $this->cartHelper = $cartHelper;
+        $this->url = $url;
+        $this->backendUrl = $backendUrl;
+        $this->stockRegistry = $stockRegistry;
+        $this->sessionHelper = $sessionHelper;
+    }
+
+    /**
+     * Pre-Auth hook: Create order.
+     *
+     * @api
+     *
+     * @param string $type - "order.create"
+     * @param array $order
+     * @param string $currency
+     *
+     * @return void
+     * @throws \Exception
```

Comment:
looks like you have big try-catch here, does this method actually throws exception?

Reply:
removed